src/gun_http2.erl

```diff
@@ -341,13 +344,15 @@ maybe_ack_or_notify(State=#http2_state{reply_to=ReplyTo, socket=Socket,
 				_ ->
 					ok
 			end,
-			Transport:send(Socket, cow_http2:settings_ack());
+			Transport:send(Socket, cow_http2:settings_ack()),
+			State;
 		{ping, Opaque} ->
 			Transport:send(Socket, cow_http2:ping_ack(Opaque));
+		{ping_ack, _Opaque} ->
+			State#http2_state{pings_sent = 0};
```

Comment:
That's probably wrong, it should be Current-1.

Reply:
Yes, makes more sense to accept a delay but eventually all pings need to be acked.

Reply:
That's not what I mean. If you send 3 pings and then receive a ping_ack, pings_sent will go from 3 to 0, and then when you send more pings and receive ping_acks you're out of sync with reality.

Reply:
Ah, right. That can make us keep a connection with even if it has a very long network delay.

My interpretation of the requirement was that we need to receive one ack for every N pings sent, so the increment and reset achieves this. We could achieve this with a different payload for each ping and match each ack to its ping. But I don't think the extra bookkeeping is worth it and furthermore there is no reason to accept data loss.

Reply:
I am not saying we need to match the ping content (that's actually already done in the protocol code IIRC), just that the count needs to be accurate. We send one ping, we +1. We receive a ping_ack, we -1.

Reply:
I understand that and I'm saying that if we do that, it implies that we require an ack for each ping, not just one for every N pings. And that's fine.

Reply:
Right. The relevant spec is at https://datatracker.ietf.org/doc/html/rfc7540#section-6.7

```
   Receivers of a PING frame that does not include an ACK flag MUST send
   a PING frame with the ACK flag set in response, with an identical
   payload.  PING responses SHOULD be given higher priority than any
   other frame.
```

So you will get an ack for every ping.